controller/lib/pid/pid.cpp

```diff
@@ -23,14 +23,21 @@ float PID::Compute(Time now, float input, float setpoint) {
   if (!initialized_) {
     last_input_ = input;
     last_error_ = setpoint - input;
-
+    next_sample_time_ = now;
     // last call time defined as now - SampleTime to enable computation on first
     // call (user should call Compute() immediately after SetMode(Auto))
+    last_update_time_ = now - sample_period_;
     initialized_ = true;
   }
 
-  float samplesTimeChangeSec = sample_period_.seconds();
-
+  // compute actual samples time-difference to take jitter into account in
+  // integral and derivative
+  Duration effectiveSampleTime = (now - last_update_time_);
+  float samplesTimeChangeSec = effectiveSampleTime.seconds();
+  // condition to update output : 1 sample time has passed and we have new data
+  if (now < next_sample_time_ || samplesTimeChangeSec <= 0) {
+    return last_output_;
+  }
```

Comment:
When called periodically with some jitter you're going to lose a loop update at some point then your sample time will be off by approximately 1 loop period on every subsequent call.  Not sure what you hope to gain by this.

Reply:
This is merely part 1 of my evil plan.

Part 2 of the evil plan is to increase the timer granularity.  Thank you for catching my error in that PR.

Part 3 of my evil plan is https://github.com/RespiraWorks/VentilatorSoftware/pull/467, which I think speaks to your concern.